stdlib/unittest/case.pyi

```diff
@@ -275,7 +281,7 @@ class _AssertLogsContext:
     records: List[logging.LogRecord]
     output: List[str]
     def __init__(self, test_case: TestCase, logger_name: str, level: int) -> None: ...
-    def __enter__(self) -> _AssertLogsContext: ...
+    def __enter__(self) -> _LoggingWatcher | None: ...
```

Comment:
Returning `None` is new in Python 3.10:

```suggestion
    if sys.version_info >= (3, 10):
        def __enter__(self) -> _LoggingWatcher | None: ...
    else:
        def __enter__(self) -> _LoggingWatcher: ...
```

Relevant source codes:

```
akuli@akuli-desktop:~$ python3.9 bin/pysrc unittest._log._AssertLogsContext.__enter__
    def __enter__(self):
        if isinstance(self.logger_name, logging.Logger):
            logger = self.logger = self.logger_name
        else:
            logger = self.logger = logging.getLogger(self.logger_name)
        formatter = logging.Formatter(self.LOGGING_FORMAT)
        handler = _CapturingHandler()
        handler.setFormatter(formatter)
        self.watcher = handler.watcher
        self.old_handlers = logger.handlers[:]
        self.old_level = logger.level
        self.old_propagate = logger.propagate
        logger.handlers = [handler]
        logger.setLevel(self.level)
        logger.propagate = False
        return handler.watcher
akuli@akuli-desktop:~$ python3.10 bin/pysrc unittest._log._AssertLogsContext.__enter__
    def __enter__(self):
        if isinstance(self.logger_name, logging.Logger):
            logger = self.logger = self.logger_name
        else:
            logger = self.logger = logging.getLogger(self.logger_name)
        formatter = logging.Formatter(self.LOGGING_FORMAT)
        handler = _CapturingHandler()
        handler.setLevel(self.level)
        handler.setFormatter(formatter)
        self.watcher = handler.watcher
        self.old_handlers = logger.handlers[:]
        self.old_level = logger.level
        self.old_propagate = logger.propagate
        logger.handlers = [handler]
        logger.setLevel(self.level)
        logger.propagate = False
        if self.no_logs:
            return
        return handler.watcher
```

Reply:
sounds good